buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt

```diff
@@ -63,17 +66,37 @@ open class BuildScanPlugin : Plugin<Project> {
 
         if (isCiServer && !isTravis) {
             extractAllReportsFromCI()
+            monitorUnexpectedCacheMisses()
         }
 
         extractCheckstyleAndCodenarcData()
         extractBuildCacheData()
     }
 
+    private
+    fun Project.monitorUnexpectedCacheMisses() {
+        gradle.taskGraph.afterTask {
+            if (!state.skipped && !isExcludedBuild() && this.isMonitored()) {
+                buildScan.tag("CACHE_MISS")
+            }
+        }
+    }
+
+    private
+    fun Task.isMonitored() =
+        this is AbstractCompile || this is ClasspathManifest
+
+    private
+    fun Project.isExcludedBuild() =
+        // Two expected cache-miss:
+        // 1. compileAll is seed build
+        // 2. Gradleexception is building on Java 8
```

Comment:
The point is that Gradleception is re-building Gradle with a new Gradle version. And we don't cache between Gradle versions.

Note also that it is `Gradleception`, not `Gradleexception`.